spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -764,6 +776,7 @@ protected void doStart() throws Exception {
 						+ Arrays.asList(queueNames));
 			}
 		}
+		checkMismatchedQueues();
```

Comment:
Let me share my mind here with you!
From the beginning I thought that we should move this call after the `this.rabbitAdmin` initialization. But after that I noticed that we don't call its `afterPropertiesSet()`.
So, according your documentation note about admin injection to restrict declaration by the `declared-by`, it looks like we don't need move it there, because we rely on the shared `RabbitAdmin` bean for the provided `ConnectionFactory`.
The same happens even in the `redeclareElementsIfNecessary()`.

Correct?